# Profitble free Android and iOS apps

## 1.Introduction :

- In our company we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means our revenue for any given app is mostly influenced by the number of users who use our app. 
- Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.

## 2.Opening and exploring the data :

In [1]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') 
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [2]:
from csv import reader

opened_file = open('googleplaystore.csv', encoding='utf8')
read_file = reader(opened_file)
google_apps = list(read_file)
google_apps_head = google_apps[0]
google_apps = google_apps[1:]

opened_file = open('AppleStore.csv', encoding='utf8')
read_file = reader(opened_file)
apple_apps = list(read_file)
apple_apps_head = apple_apps[0]
apple_apps = apple_apps[1:][1:]

In [3]:
explore_data(google_apps,0,5,rows_and_columns=True)
explore_data(apple_apps,0,5,rows_and_columns=True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 10841
Number of columns: 13
['2', '281796108', 'Evernote - stay

In [4]:
print(apple_apps_head,'\n', google_apps_head)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'] 
 ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


- links to the describtion of the columns names for the apple_apps datatset : [apple](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home)

## 3.Data Cleaning (wrong data for entry 10472) :

In [5]:
print(google_apps[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


- The rating is 19. and the maximum rating for a Google Play app is 5, so we delete this row .

In [6]:
del google_apps[10472]
print(google_apps_head)
print(google_apps[10472])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


## 4.Cleaning Data (duplicates entries):

In [7]:
for row in google_apps :
    if row[0] == 'Instagram':
        print(row)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


- The difference between the duplicates is the number of reviews.(certainly because data was collected at different time).

In [8]:
duplicate = [] 
unique = [] 
for row in google_apps:
    if row[0] in unique :
        duplicate.append(row[0])
    else :
        unique.append(row[0])
print(len(duplicate))

1181


- In the entries for Instagram we can build a criterion for removing the duplicates. The higher the number of reviews, the more recent the data should be. Rather than removing duplicates randomly, we'll only keep the row with the highest number of reviews and remove the other entries.
- let's check few duplicate to see if the difference is only on number of reviews. 

In [9]:
for dup in duplicate[:5]:
    for row in google_apps:
        if row[0] == dup :
            print(row,'\n')

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up'] 

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up'] 

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80804', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up'] 

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device'] 

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device'] 

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,

- Looks like the difference between duplicated entries is only on the number of reviews.

## 5.Data Cleaning (Removing the duplicate entries) :

In [10]:
reviews_max = {}
for row in google_apps:
    name = row[0]
    n_reviews = float(row[3])
    if (name in reviews_max) and (reviews_max[name] < n_reviews):
        reviews_max[name] = n_reviews
    elif (name not in reviews_max):
        reviews_max[name] = n_reviews
print(len(reviews_max))

9659


In [11]:
android_clean = []
already_added = []
for row in google_apps :
    name = row[0]
    n_reviews = float(row[3])
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(row)
        already_added.append(name)
print(len(android_clean))
for row in android_clean :
    if row[0] == 'Instagram':
        print(row)

9659
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


## 6.7.Data Cleaning (removing app whose name contains a symbol that is not commonly used in English text)

In [12]:
def english_ch (s):
    i = 0
    for ch in s:
        if ord(ch) > 127 :
            i += 1
    if i > 3 :
        return False 
    else :
        return True
print(english_ch('Instagram'))
print(english_ch('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_ch('Docs To Go™ Free Office Suite'))
print(english_ch('Instachat 😜'))


True
False
True
True


In [13]:
android_apps = []
ios_apps = []
for app in android_clean :
    name = app[0]
    if english_ch(name) :
        android_apps.append(app)
print(len(android_apps))
for app in apple_apps:
    name = app[2]
    if english_ch(name):
        ios_apps.append(app)
print(len(ios_apps))

9614
6182


## 8.Data Cleaning (Isolating the free apps):

In [14]:
android_final = []
ios_final = []
for app in android_apps :
    if app[7] == '0':
        android_final.append(app)
print(len(android_final))
for app in ios_apps :
    if app[5] == '0':
        ios_final.append(app)
print(len(ios_final))

8864
3222


## 9.Most common genres for each market:

- As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

- To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

    - Build a minimal Android version of the app, and add it to Google Play.
    - If the app has a good response from users, we then develop it further.
    - If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.
- Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

- Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

In [15]:
print(google_apps_head)
print(apple_apps_head)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


- we'll need to build a frequency table for the prime_genre column of the App Store data set, and for the Genres and Category columns of the Google Play data set.

## 10.Generating Frequency Tables :

In [16]:
def freq_table(dataset, index):
    freq = {}
    for row in dataset:
        if row[index] in freq:
            freq[row[index]] += 100/len(dataset)
        else :
            freq[row[index]] = 100/len(dataset)
    return freq

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

## 11.Analyzing Frequency tables :

 - We start examining the prime_genre column of the App Store DATA  set 

In [18]:
display_table(ios_final, 12)

Games : 58.1626319056464
Entertainment : 7.883302296710134
Photo & Video : 4.965859714463075
Education : 3.6623215394165176
Social Networking : 3.2898820608317867
Shopping : 2.6070763500931133
Utilities : 2.5139664804469306
Sports : 2.1415270018621997
Music : 2.048417132216017
Health & Fitness : 2.0173805090006227
Productivity : 1.7380509000620747
Lifestyle : 1.5828677839851035
News : 1.3345747982619496
Travel : 1.2414649286157668
Finance : 1.1173184357541899
Weather : 0.8690254500310364
Food & Drink : 0.8069522036002481
Reference : 0.558659217877095
Business : 0.5276225946617009
Book : 0.4345127250155184
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


- More than half of the free english apps (58.16%) are games, Entertainment apps represents aproximatively 8%, Photo & Video close to 5%, followed by Education (3.7%) and Social Networking(3.3%) the rest of genres less than 3%. 
- The general impression is that most of the apps in apple store(more than 78%) are designed for fun (games, entertainment, photo and video, social networking, sport, music , etc), while apps designed for practical purposes (education, shopping utilities, productivity, lifestyle, etc) are more rare. however it doesn't imply that fun apps have the greatest number of users. 

- Let's continue with examining the Category and Genres column of the google play data set.

In [19]:
display_table(android_final, 1)

FAMILY : 18.907942238266926
GAME : 9.724729241877363
TOOLS : 8.46119133574016
BUSINESS : 4.591606498194979
LIFESTYLE : 3.90342960288811
PRODUCTIVITY : 3.8921480144404565
FINANCE : 3.7003610108303455
MEDICAL : 3.5311371841155417
SPORTS : 3.3957581227436986
PERSONALIZATION : 3.3167870036101235
COMMUNICATION : 3.2378158844765483
HEALTH_AND_FITNESS : 3.079873646209398
PHOTOGRAPHY : 2.944494584837555
NEWS_AND_MAGAZINES : 2.7978339350180583
SOCIAL : 2.6624548736462152
TRAVEL_AND_LOCAL : 2.335288808664261
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.14350180505415
DATING : 1.861462093862813
VIDEO_PLAYERS : 1.7937725631768928
MAPS_AND_NAVIGATION : 1.398916967509025
FOOD_AND_DRINK : 1.2409747292418778
EDUCATION : 1.1620036101083042
ENTERTAINMENT : 0.9589350180505433
LIBRARIES_AND_DEMO : 0.9363718411552363
AUTO_AND_VEHICLES : 0.9250902527075828
HOUSE_AND_HOME : 0.8235559566787015
WEATHER : 0.8009927797833946
EVENTS : 0.7107400722021667
PARENTING : 0.6543321299638993
ART_AND_DESIGN : 0.6

- Almost 19% of the free english apps in google play are family apps, games represents 9.7%, followed by tools (8.4%), buisness(4.6%) and lifestyle(3.9%) comes respectively fourth and fifth.
- Each of those apps :(PRODUCTIVITY, FINANCE, MEDICAL, SPORTS, PERSONALIZATION, COMMUNICATION, HEALTH_AND_FITNESS, PHOTOGRAPHY) represents aproximatively 3%
- The Apps designed for practical purposes are better represented in GOOGLE Play compared to App Store. 

In [20]:
display_table(android_final, 9)

Tools : 8.449909747292507
Entertainment : 6.069494584837599
Education : 5.34747292418777
Business : 4.591606498194979
Productivity : 3.8921480144404565
Lifestyle : 3.8921480144404565
Finance : 3.7003610108303455
Medical : 3.5311371841155417
Sports : 3.46344765342962
Personalization : 3.3167870036101235
Communication : 3.2378158844765483
Action : 3.1024368231047053
Health & Fitness : 3.079873646209398
Photography : 2.944494584837555
News & Magazines : 2.7978339350180583
Social : 2.6624548736462152
Travel & Local : 2.3240072202166075
Shopping : 2.2450361010830324
Books & Reference : 2.14350180505415
Simulation : 2.041967509025268
Dating : 1.861462093862813
Arcade : 1.8501805054151597
Video Players & Editors : 1.771209386281586
Casual : 1.7599277978339327
Maps & Navigation : 1.398916967509025
Food & Drink : 1.2409747292418778
Puzzle : 1.1281588447653441
Racing : 0.9927797833935037
Role Playing : 0.9363718411552363
Libraries & Demo : 0.9363718411552363
Auto & Vehicles : 0.9250902527075828


- The difference between the Category and Genres column is not clear, but we can notice that the Genre column has more categories. 
- For now, we found that the App store is dominated by apps designed for fun, while Google Play have a balanced quantity of both practical apps and for-fun apps.
- let's continue exploring and examining the datasets to get an idea about the genres that have most users.

## 12. Calculating the average number of user ratings per app genre on the App Store:

In [28]:
unique_genres = freq_table(ios_final, 12)
avg_ratings = {}
for genre in unique_genres :
    total = 0
    len_genre = 0 
    for app in ios_final :
        if app[12] == genre :
            total += float(app[6])
            len_genre += 1
    avg_ratings[genre] = total / len_genre
avg_ratings

{'Productivity': 21028.410714285714,
 'Weather': 52279.892857142855,
 'Shopping': 26919.690476190477,
 'Reference': 74942.11111111111,
 'Finance': 31467.944444444445,
 'Music': 57326.530303030304,
 'Utilities': 18684.456790123455,
 'Travel': 28243.8,
 'Social Networking': 71548.34905660378,
 'Sports': 23008.898550724636,
 'Health & Fitness': 23298.015384615384,
 'Games': 22788.6696905016,
 'Food & Drink': 33333.92307692308,
 'News': 21248.023255813954,
 'Book': 39758.5,
 'Photo & Video': 28441.54375,
 'Entertainment': 14029.830708661417,
 'Business': 7491.117647058823,
 'Lifestyle': 16485.764705882353,
 'Education': 7003.983050847458,
 'Navigation': 86090.33333333333,
 'Medical': 612.0,
 'Catalogs': 4004.0}

- The first five genres with the gratest average number of user ratings : 
    - Navigation        : 86090.33
    - Reference         : 74942.11
    - Social Networking : 71548.35
    - Music             : 57326.53
    - Weather           : 52279.89
- let's go deep in each of the genres above. 

In [29]:
print('Navigation')
for app in ios_final:
    if app[12] == 'Navigation':
        print(app[2],', user_ratings : ',app[6])

Navigation
Waze - GPS Navigation, Maps & Real-time Traffic , user_ratings :  345046
Geocaching® , user_ratings :  12811
ImmobilienScout24: Real Estate Search in Germany , user_ratings :  187
Railway Route Search , user_ratings :  5
CoPilot GPS – Car Navigation & Offline Maps , user_ratings :  3582
Google Maps - Navigation & Transit , user_ratings :  154911


- Average of user ratings for Navigation apps is heavily influenced by Waze and Google Maps.(both represent aproximatively one half milion of user ratings)

In [30]:
print('Reference')
for app in ios_final:
    if app[12] == 'Reference':
        print(app[2],', user_ratings : ',app[6])

Reference
Bible , user_ratings :  985920
Dictionary.com Dictionary & Thesaurus , user_ratings :  200047
Dictionary.com Dictionary & Thesaurus for iPad , user_ratings :  54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran , user_ratings :  18418
Merriam-Webster Dictionary , user_ratings :  16849
Google Translate , user_ratings :  26786
Night Sky , user_ratings :  12122
WWDC , user_ratings :  762
Jishokun-Japanese English Dictionary & Translator , user_ratings :  0
教えて!goo , user_ratings :  0
VPN Express , user_ratings :  14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition , user_ratings :  17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools , user_ratings :  4693
Guides for Pokémon GO - Pokemon GO News and Cheats , user_ratings :  826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free , user_ratings :  718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket E

- Same for Reference genres, user ratings is influenced by Bible and Dictionary.com Dictionary & Thesaurus.
- it seems that we can turn a famous book to an app with intresting features (Audio version, daily quotes ...).and on top of that, we have more opportunities in practical apps compared to for-fun apps.

In [31]:
print('Social Networking')
for app in ios_final:
    if app[12] == 'Social Networking':
        print(app[2],', user_ratings : ',app[6])

Social Networking
Facebook , user_ratings :  2974676
LinkedIn , user_ratings :  71856
Skype for iPhone , user_ratings :  373519
Tumblr , user_ratings :  334293
Match™ - #1 Dating App. , user_ratings :  60659
WhatsApp Messenger , user_ratings :  287589
TextNow - Unlimited Text + Calls , user_ratings :  164963
Grindr - Gay and same sex guys chat, meet and date , user_ratings :  23201
imo video calls and chat , user_ratings :  18841
Ameba , user_ratings :  269
Weibo , user_ratings :  7265
Badoo - Meet New People, Chat, Socialize. , user_ratings :  34428
Kik , user_ratings :  260965
Qzone , user_ratings :  1649
Fake-A-Location Free ™ , user_ratings :  354
Tango - Free Video Call, Voice and Chat , user_ratings :  75412
MeetMe - Chat and Meet New People , user_ratings :  97072
SimSimi , user_ratings :  23530
Viber Messenger – Text & Call , user_ratings :  164249
Find My Family, Friends & iPhone - Life360 Locator , user_ratings :  43877
Weibo HD , user_ratings :  16772
POF - Best Dating App f

-  The average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. We can have a small shot if we sort out an app with innovating features and ideas(For exemple : an ID control of subscribers to reduce fake profiles) but it still hard to challenge those giants !

In [32]:
print('Music')
for app in ios_final:
    if app[12] == 'Music':
        print(app[2],', user_ratings : ',app[6])

Music
Pandora - Music & Radio , user_ratings :  1126879
Shazam - Discover music, artists, videos & lyrics , user_ratings :  402925
iHeartRadio – Free Music & Radio Stations , user_ratings :  293228
Deezer - Listen to your Favorite Music & Playlists , user_ratings :  4677
Sonos Controller , user_ratings :  48905
NRJ Radio , user_ratings :  38
radio.de - Der Radioplayer , user_ratings :  64
Spotify Music , user_ratings :  878563
SoundCloud - Music & Audio , user_ratings :  135744
Sing Karaoke Songs Unlimited with StarMaker , user_ratings :  26227
SoundHound Song Search & Music Player , user_ratings :  82602
Ringtones for iPhone & Ringtone Maker , user_ratings :  25403
Coach Guitar - Lessons & Easy Tabs For Beginners , user_ratings :  2416
TuneIn Radio - MLB NBA Audiobooks Podcasts Music , user_ratings :  110420
Magic Piano by Smule , user_ratings :  131695
QQ音乐HD , user_ratings :  224
The Singing Machine Mobile Karaoke App , user_ratings :  130
Bandsintown Concerts , user_ratings :  3084

- Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

- Weather apps : people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

## 13. Google Play genres's popularity : 

In [33]:
unique_category = freq_table(android_final, 1)
avg_installs = {}
for category in unique_category :
    total = 0 
    len_category = 0
    for app in android_final :
        if app[1] == category :
            installs = app[5]
            installs = installs.replace('+', '')
            installs = installs.replace(',', '')
            total += float(installs)
            len_category += 1
    avg_installs[category] = total / len_category
avg_installs

{'ART_AND_DESIGN': 1986335.0877192982,
 'AUTO_AND_VEHICLES': 647317.8170731707,
 'BEAUTY': 513151.88679245283,
 'BOOKS_AND_REFERENCE': 8767811.894736841,
 'BUSINESS': 1712290.1474201474,
 'COMICS': 817657.2727272727,
 'COMMUNICATION': 38456119.167247385,
 'DATING': 854028.8303030303,
 'EDUCATION': 1833495.145631068,
 'ENTERTAINMENT': 11640705.88235294,
 'EVENTS': 253542.22222222222,
 'FINANCE': 1387692.475609756,
 'FOOD_AND_DRINK': 1924897.7363636363,
 'HEALTH_AND_FITNESS': 4188821.9853479853,
 'HOUSE_AND_HOME': 1331540.5616438356,
 'LIBRARIES_AND_DEMO': 638503.734939759,
 'LIFESTYLE': 1437816.2687861272,
 'GAME': 15588015.603248259,
 'FAMILY': 3695641.8198090694,
 'MEDICAL': 120550.61980830671,
 'SOCIAL': 23253652.127118643,
 'SHOPPING': 7036877.311557789,
 'PHOTOGRAPHY': 17840110.40229885,
 'SPORTS': 3638640.1428571427,
 'TRAVEL_AND_LOCAL': 13984077.710144928,
 'TOOLS': 10801391.298666667,
 'PERSONALIZATION': 5201482.6122448975,
 'PRODUCTIVITY': 16787331.344927534,
 'PARENTING': 5426

- The first nine genres with the gratest average number of installs : 
    - COMMUNICATION      : 38 456 119.16
    - VIDEO_PLAYERS      : 24 727 872.45
    - SOCIAL             : 23 253 652.12
    - PHOTOGRAPHY        : 17 840 110.40
    - PRODUCTIVITY       : 16 787 331.34
    - GAME               : 15 588 015.60
    - TRAVEL_AND_LOCAL   : 13 984 077.71
    - ENTERTAINMENT      : 11 640 705.88
    - TOOLS              : 10 801 391.29
    - NEWS_AND_MAGAZINES :  9 549 178.46
    - BOOKS_AND_REFERENCE:  8 767 811.89
- let's go deep in each of the genres above. 

In [34]:
print('COMMUNICATION')
for app in android_final:
    if app[1] == 'COMMUNICATION':
        print(app[0],', installs : ', app[5])

COMMUNICATION
WhatsApp Messenger , installs :  1,000,000,000+
Messenger for SMS , installs :  10,000,000+
My Tele2 , installs :  5,000,000+
imo beta free calls and text , installs :  100,000,000+
Contacts , installs :  50,000,000+
Call Free – Free Call , installs :  5,000,000+
Web Browser & Explorer , installs :  5,000,000+
Browser 4G , installs :  10,000,000+
MegaFon Dashboard , installs :  10,000,000+
ZenUI Dialer & Contacts , installs :  10,000,000+
Cricket Visual Voicemail , installs :  10,000,000+
TracFone My Account , installs :  1,000,000+
Xperia Link™ , installs :  10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard , installs :  10,000,000+
Skype Lite - Free Video Call & Chat , installs :  5,000,000+
My magenta , installs :  1,000,000+
Android Messages , installs :  100,000,000+
Google Duo - High Quality Video Calls , installs :  500,000,000+
Seznam.cz , installs :  1,000,000+
Antillean Gold Telegram (original version) , installs :  100,000+
AT&T Visual Voicemail , in

In [35]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '100,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '1,000,000,000+'):
        print(app[0],', installs : ', app[5])

WhatsApp Messenger , installs :  1,000,000,000+
imo beta free calls and text , installs :  100,000,000+
Android Messages , installs :  100,000,000+
Google Duo - High Quality Video Calls , installs :  500,000,000+
Messenger – Text and Video Chat for Free , installs :  1,000,000,000+
imo free video calls and chat , installs :  500,000,000+
Skype - free IM & video calls , installs :  1,000,000,000+
Who , installs :  100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji , installs :  100,000,000+
LINE: Free Calls & Messages , installs :  500,000,000+
Google Chrome: Fast & Secure , installs :  1,000,000,000+
Firefox Browser fast & private , installs :  100,000,000+
UC Browser - Fast Download Private & Secure , installs :  500,000,000+
Gmail , installs :  1,000,000,000+
Hangouts , installs :  1,000,000,000+
Messenger Lite: Free Calls & Messages , installs :  100,000,000+
Kik , installs :  100,000,000+
KakaoTalk: Free Calls & Text , installs :  100,000,000+
Opera Mini - fast web browser , i

- On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs.

In [36]:
for app in android_final:
    if app[1] == 'SOCIAL' and (app[5] == '100,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '1,000,000,000+'):
        print(app[0],', installs : ', app[5])

Facebook , installs :  1,000,000,000+
Facebook Lite , installs :  500,000,000+
Tumblr , installs :  100,000,000+
Pinterest , installs :  100,000,000+
Google+ , installs :  1,000,000,000+
Badoo - Free Chat & Dating App , installs :  100,000,000+
Tango - Live Video Broadcast , installs :  100,000,000+
Instagram , installs :  1,000,000,000+
Snapchat , installs :  500,000,000+
LinkedIn , installs :  100,000,000+
Tik Tok - including musical.ly , installs :  100,000,000+
BIGO LIVE - Live Stream , installs :  100,000,000+
VK , installs :  100,000,000+


In [37]:
for app in android_final:
    if app[1] == 'VIDEO_PLAYERS' and (app[5] == '100,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '1,000,000,000+'):
        print(app[0],', installs : ', app[5])

YouTube , installs :  1,000,000,000+
Motorola Gallery , installs :  100,000,000+
VLC for Android , installs :  100,000,000+
Google Play Movies & TV , installs :  1,000,000,000+
MX Player , installs :  500,000,000+
Dubsmash , installs :  100,000,000+
VivaVideo - Video Editor & Photo Movie , installs :  100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera , installs :  100,000,000+
Motorola FM Radio , installs :  100,000,000+


In [38]:
for app in android_final:
    if app[1] == 'PHOTOGRAPHY' and (app[5] == '100,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '1,000,000,000+'):
        print(app[0],', installs : ', app[5])

B612 - Beauty & Filter Camera , installs :  100,000,000+
YouCam Makeup - Magic Selfie Makeovers , installs :  100,000,000+
Sweet Selfie - selfie camera, beauty cam, photo edit , installs :  100,000,000+
Google Photos , installs :  1,000,000,000+
Retrica , installs :  100,000,000+
Photo Editor Pro , installs :  100,000,000+
BeautyPlus - Easy Photo Editor & Selfie Camera , installs :  100,000,000+
PicsArt Photo Studio: Collage Maker & Pic Editor , installs :  100,000,000+
Photo Collage Editor , installs :  100,000,000+
Z Camera - Photo Editor, Beauty Selfie, Collage , installs :  100,000,000+
PhotoGrid: Video & Pic Collage Maker, Photo Editor , installs :  100,000,000+
Candy Camera - selfie, beauty camera, photo editor , installs :  100,000,000+
YouCam Perfect - Selfie Photo Editor , installs :  100,000,000+
Camera360: Selfie Photo Editor with Funny Sticker , installs :  100,000,000+
S Photo Editor - Collage Maker , Photo Collage , installs :  100,000,000+
AR effect , installs :  100,000

In [39]:
for app in android_final:
    if app[1] == 'PRODUCTIVITY' and (app[5] == '100,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '1,000,000,000+'):
        print(app[0],', installs : ', app[5])

Microsoft Word , installs :  500,000,000+
Microsoft Outlook , installs :  100,000,000+
Microsoft OneDrive , installs :  100,000,000+
Microsoft OneNote , installs :  100,000,000+
Google Keep , installs :  100,000,000+
ES File Explorer File Manager , installs :  100,000,000+
Dropbox , installs :  500,000,000+
Google Docs , installs :  100,000,000+
Microsoft PowerPoint , installs :  100,000,000+
Samsung Notes , installs :  100,000,000+
SwiftKey Keyboard , installs :  100,000,000+
Google Drive , installs :  1,000,000,000+
Adobe Acrobat Reader , installs :  100,000,000+
Google Sheets , installs :  100,000,000+
Microsoft Excel , installs :  100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet , installs :  100,000,000+
Google Slides , installs :  100,000,000+
ColorNote Notepad Notes , installs :  100,000,000+
Evernote – Organizer, Planner for Notes & Memos , installs :  100,000,000+
Google Calendar , installs :  500,000,000+
Cloud Print , installs :  500,000,000+
CamScanner - Phone

- We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).
- let's continue by examining the game genre :

In [40]:
for app in android_final:
    if app[1] == 'GAME' and (app[5] == '100,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '1,000,000,000+'):
        print(app[0],', installs : ', app[5])

Sonic Dash , installs :  100,000,000+
PAC-MAN , installs :  100,000,000+
Roll the Ball® - slide puzzle , installs :  100,000,000+
Piano Tiles 2™ , installs :  100,000,000+
Pokémon GO , installs :  100,000,000+
Extreme Car Driving Simulator , installs :  100,000,000+
Trivia Crack , installs :  100,000,000+
Angry Birds 2 , installs :  100,000,000+
Candy Crush Saga , installs :  500,000,000+
8 Ball Pool , installs :  100,000,000+
Subway Surfers , installs :  1,000,000,000+
Candy Crush Soda Saga , installs :  100,000,000+
Clash Royale , installs :  100,000,000+
Clash of Clans , installs :  100,000,000+
Plants vs. Zombies FREE , installs :  100,000,000+
Pou , installs :  500,000,000+
Flow Free , installs :  100,000,000+
My Talking Angela , installs :  100,000,000+
slither.io , installs :  100,000,000+
Cooking Fever , installs :  100,000,000+
Yes day , installs :  100,000,000+
Score! Hero , installs :  100,000,000+
Dream League Soccer 2018 , installs :  100,000,000+
My Talking Tom , installs

- The game genre seems satured, and unless an innovating idea it's better to focus on others genres. 
- Since we are looking fo app profile with potential in both App Store and Google Play.
- let's look at the BOOKS_AND_REFERENCE genres :

In [41]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '100,000,000+'
                                     or app[5] == '500,000,000+'
                                     or app[5] == '1,000,000,000+'):
        print(app[0],', installs : ', app[5])

Google Play Books , installs :  1,000,000,000+
Bible , installs :  100,000,000+
Amazon Kindle , installs :  100,000,000+
Wattpad 📖 Free Books , installs :  100,000,000+
Audiobooks from Audible , installs :  100,000,000+


- There are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity

In [42]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0],', installs : ', app[5])

Wikipedia , installs :  10,000,000+
Cool Reader , installs :  10,000,000+
Book store , installs :  1,000,000+
FBReader: Favorite Book Reader , installs :  10,000,000+
Free Books - Spirit Fanfiction and Stories , installs :  1,000,000+
AlReader -any text book reader , installs :  5,000,000+
FamilySearch Tree , installs :  1,000,000+
Cloud of Books , installs :  1,000,000+
ReadEra – free ebook reader , installs :  1,000,000+
Ebook Reader , installs :  5,000,000+
Read books online , installs :  5,000,000+
eBoox: book reader fb2 epub zip , installs :  1,000,000+
All Maths Formulas , installs :  1,000,000+
Ancestry , installs :  5,000,000+
HTC Help , installs :  10,000,000+
Moon+ Reader , installs :  10,000,000+
English-Myanmar Dictionary , installs :  1,000,000+
Golden Dictionary (EN-AR) , installs :  1,000,000+
All Language Translator Free , installs :  1,000,000+
Aldiko Book Reader , installs :  10,000,000+
Dictionary - WordWeb , installs :  5,000,000+
50000 Free eBooks & Free AudioBooks

- Same as in the App Store : it seems that we can turn a famous book to an app, however it looks like the market is already full of librariesso we will have to add intresting features (Audio version, daily quotes ...).

# Conclusion:

- In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

- We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.